query-engine/connector-test-kit-rs/query-engine-tests/tests/queries/simple/find_unique.rs

```diff
@@ -89,4 +89,141 @@ mod find_unique {
 
         Ok(())
     }
+
+    fn simple_uniq_idx_with_embedded() -> String {
+        indoc! {r#"
+        type Location {
+            address String
+        }
+
+        model A {
+            #id(id, Int, @id)
+            name String
+            location Location
+
+            @@unique([location.address])
+        }
+        "#}
+        .to_string()
+    }
+
+    #[connector_test(schema(simple_uniq_idx_with_embedded), only(MongoDb))]
+    async fn simple_embedded_type(runner: Runner) -> TestResult<()> {
+        run_query!(
+            runner,
+            indoc! {r#"mutation {
+            createManyA(data: [
+                {id: 1 name: "foo" location: {set: {address: "a"}}},
+                {id: 2 name: "foo" location: {set: {address: "b"}}},
+                {id: 3 name: "foo" location: {set: {address: "c"}}},
+            ]) { count }
+        }"#}
+        );
+
+        assert_query!(
+            runner,
+            r#"query { findUniqueA(where: { 
+                location_address: {
+                    location: {
+                        address: "a"
+                    }
+                }
+            }) { id }}"#,
+            r#"{"data":{"findUniqueA":{"id":1}}}"#
+        );
+
+        Ok(())
+    }
+
+    fn composite_uniq_idx_with_embedded() -> String {
```

Comment:
These tests seem to test the happy path. Could you see if there are other tests worth adding to make sure we have a really good comprehensive set of tests. For example:
1. Multiple uniques on a model
2. A unique with a composite that has a lot more depth `location.address.street_number`
3. Some errors that we would expect if this fails